### DISCUSSION TITLE
How to render Component from Views

### DISCUSSION THREAD

**Discussion by itsyoshio** [yippee-fun/phlex#517] [Help] [answered]
When trying the [Rails Example](https://www.phlex.fun/rails/generators/) i just assumed i could do something like 

```ruby
class Articles::IndexView < ApplicationView
  def template
    h1 { "Articles index" }
    p { "Find me in app/views/articles/index_view.rb" }
    render Card.new # < render the Card Component in the Template, or anywhere else.
  end
end
```

but i came across `uninitialized constant Articles::IndexView::Card`.

I looked at Code on Github (Hottable and [fly-phlex-wired](https://github.com/karn18/fly-phlex-wired/blob/d98a3872178a683931e189d31dd1718255a49b78/app/views/layouts/application.html.erb)) but i can't quite grasp how Components are meant to be used.

Any Guide would be appreciated!

**Comment by willcosgrove** [comment, marked as answer]
That should work, provided you have a `Card` phlex component that is defined and accessible from that namespace.

Just looking at the guide that you linked to, it creates a `CardComponent` class (as opposed to the `Card` class referenced in your code). Did you mean to have that render line rendering a `CardComponent`?

```ruby
class Articles::IndexView < ApplicationView
  def template
    h1 { "Articles index" }
    p { "Find me in app/views/articles/index_view.rb" }
    render CardComponent.new
  end
end
```

**Comment by joeldrapper** [comment]
If you've run the install generator, you should have `app/views/components` in your autoload paths. If you define `CardComponent` in `app/views/components/card_component.rb`, you can render it from a view with `render CardComponent.new`.

If you want to call the component `Card` instead of `CardComponent`, you would need to also rename the file to `app/views/components/card.rb` so it's auto-loaded correctly by Rails.

Also, the HotTable repo is outdated, so I wouldn't use that as a reference.
